hugolib/pageSort_test.go

```diff
@@ -113,6 +113,26 @@ func TestPageSortReverse(t *testing.T) {
 	assert.True(t, probablyEqualPages(p2, p1.Reverse()))
 }
 
+func TestPageSortByParam(t *testing.T) {
+	unsorted := createSortTestPages(10)
+	firstValue, _ := unsorted[0].Param("arbitrary")
```

Comment:
You're missing the test case where the param isn't set. I would say that those pages should come at the end.